public/js/ui.js

```diff
@@ -7,19 +7,27 @@ let inferenceText;
 let collectionText;
 
 // All handlers are private
-function toggleCollection() {
-  state.isCollectionOn = !state.isCollectionOn;
-  collectionButton.innerHTML = state.isCollectionOn ? 'On' : 'Off';
-}
-
-function toggleInference() {
-  state.isInferenceOn = !state.isInferenceOn;
-  inferenceButton.innerHTML = state.isInferenceOn ? 'On' : 'Off';
+function toggleButton(button, flag) {
```

Comment:
I see that this function isn't new, but do you think we should start writing ES6 functions? At least once we touch existing functions, we might as well

We have to look at that bundler and babel task soon, I'll get a start on that actually

Reply:
Also really nit-picky and not important - what do you think about `state` rather than `flag` here?

Reply:
> I see that this function isn't new, but do you think we should start writing ES6 functions? At least once we touch existing functions, we might as well

Yeah, I normally mostly use arrow functions (and prefer them to function declaration) and I was considering using them for the new/replaced functions but felt it might be inconsistent with what @misterpeddy wrote.
@misterpeddy Do you think we should stick to function declaration (ES5-) or switch to (when suitable) function expressions/arrow functions (ES6)?

> We have to look at that bundler and babel task soon, I'll get a start on that actually

Yeah, I was actually considering trying something out with Rollup (which I'm curious to try).
As for Babel, I think it's definitely a good idea so we can use ES9-10 without worrying about browser support (given that now ES6/ES5 has overall support of ~98%/100%).

> Also really nit-picky and not important - what do you think about `state` rather than `flag` here?

Well, I'm not completely convinced I wrote that bit well, but given that the pool of values were boolean (`isCollectionOn`/`isInferenceOn`), I thought it would be the most readable and maintainable way; at least compared to one with `parameterName`/`propertyName` and it being used as:
```js
function toggleButton(button, propertyName) {
  const property = `is${propertyName}On`;
  state[property] = ...;
}
```
We could use state here but the objective of the function was to refactor the on/off toggle functions into a single one (to make it DRY) while still being used for different buttons.

If anyone has a better way than I'll be more than happy to consider it (perhaps one that will also consider the label button with boolean values).

Reply:
Thanks for pointing this out! I didn't know JS best practices prefer function expressions but just read up on it and definitely seems so + makes sense so makes sense to switch.

Regarding UI control logic, I've seen some neat frameworks that take care of some of this for you ([ex](https://github.com/dataarts/dat.gui)) - unsure of the tradeoffs of using them vs. doing it ourselves (since we'll likely have very few UI control elements anyway) - until then thanks for DRYing the implementation :) 
I think your and Amir's issue is due to my poor choice of calling the large global dump of variables `state` as opposed to `context` or something else.. technically the second argument is going to be the state that the object referenced by the first argument should transition to -- but since we're using `state` elsewhere, flag/attribute/value would all work imo.

Reply:
> Thanks for pointing this out! I didn't know JS best practices prefer function expressions but just read up on it and definitely seems so + makes sense so makes sense to switch.

As far as JS best practises goes, the choice between function declaration (`function foo() {}`) and function expression (`const foo = () => {}`) comes down to the following:
- what will `this` be expected to refer to? Lexical sense? Use arrow functions or the declarative form
- readability (is it going to be inlined as a callback or not?)
- familiarity with EcmaScript (ES) standards and personal preference

Regarding [NodeJS best practises](https://github.com/goldbergyoni/nodebestpractices) (some of which are perfectly applicable to browser JS), the consensus goes towards arrow functions (https://github.com/goldbergyoni/nodebestpractices/#-312-use-arrow-function-expressions-).

So shall we refactor those functions to arrow ones in this PR or put it in another PR or [create an issue](https://github.com/misterpeddy/hands-down/issues/new) for it?

> Regarding UI control logic, I've seen some neat frameworks that take care of some of this for you (ex) - unsure of the tradeoffs of using them vs. doing it ourselves (since we'll likely have very few UI control elements anyway) - until then thanks for DRYing the implementation :)

I've never seen such frameworks until looking at https://workshop.chromeexperiments.com/examples/gui/#1--Basic-Usage. I'm not fully sure of how Dat.GUI (or any similar frameworks) would help regarding buttons/switches that ultimately will have a set look & feel when deployed but I could see how it may help.

> I think your and Amir's issue is due to my poor choice of calling the large global dump of variables state as opposed to context or something else.. technically the second argument is going to be the state that the object referenced by the first argument should transition to -- but since we're using state elsewhere, flag/attribute/value would all work imo.

I can see why you opted to name it `state` especially when some state management libraries do that (e.g. Vuex for VueJS) so I'm personally fine with it.
I'll rename the `flag` parameter to `value` which is short and concise.